### PR TITLE
feat(text-field): Add CSS-only version of outlined text field

### LINKED ISSUE #1601: Add CSS for css only version of outlined text field
Add CSS for css only version of outlined text field

... also add a demo

### PR DESCRIPTION
**Pull request by bonniezhou**
Fixes #1601.

### CODE REVIEW

**Comment by bonniezhou** [comment]
Confirmed with designers that the outline should be darkened on hover, and changed to the primary color on focus.

File: packages/mdc-textfield/mdc-text-field.scss [hunk 1]
```diff
@@ -655,16 +655,39 @@
...
-.mdc-text-field--box:not(.mdc-text-field--upgraded) {
-  height: 56px;
+.mdc-text-field:not(.mdc-text-field--upgraded) {
+  &.mdc-text-field--outlined {
+    height: 56px;
 
-  &::before,
-  &::after {
-    border-radius: 0;
+    .mdc-text-field__input{
```
Comment (kfranqueiro) [inline packages/mdc-textfield/mdc-text-field.scss]: Missing space before `{`

File: packages/mdc-textfield/mdc-text-field.scss [hunk 2]
```diff
@@ -655,16 +655,39 @@
   }
 }
 
-.mdc-text-field--box:not(.mdc-text-field--upgraded) {
-  height: 56px;
+.mdc-text-field:not(.mdc-text-field--upgraded) {
+  &.mdc-text-field--outlined {
```
Comment (kfranqueiro) [inline packages/mdc-textfield/mdc-text-field.scss]: Is there a reason you nested these selectors and increased their specificity?

Would this work just as well as `.mdc-text-field--box:not(...)` and `.mdc-text-field--outlined:not(...)`?

Comment (bonniezhou) [reply packages/mdc-textfield/mdc-text-field.scss]: `.mdc-text-field--outlined:not(...)` wasn't strong enough to override the `.mdc-text-field:not(.mdc-text-field--upgraded):not(.mdc-text-field--textarea) .mdc-text-field__input` selector from above, so I removed `:not(.mdc-text-field--textarea)` from that and put in its own rule.

So now we have 
- `.mdc-text-field--outlined:not(...)`
- `.mdc-text-field--box:not(...)`
- `.mdc-text-field--textarea:not(...)`

File: packages/mdc-textfield/mdc-text-field.scss [hunk 3]
```diff
@@ -655,6 +653,27 @@
...
 
+.mdc-text-field--outlined:not(.mdc-text-field--upgraded) {
+  height: 56px;
+
+  .mdc-text-field__input {
+    @include mdc-text-field-outlined-corner-radius($mdc-text-field-border-radius);
+
+    height: 100%;
+    padding: 0 0 0 12px;
+    transition: mdc-text-field-transition(border);
```
Comment (kfranqueiro) [inline packages/mdc-textfield/mdc-text-field.scss]: See comment below about transitioning on border / border-color, so I'm wondering if we should forego transition on this. Depending on what your primary color is, transitioning between hover and focus can be rather subtle anyway.

File: packages/mdc-textfield/mdc-text-field.scss [hunk 4]
```diff
@@ -668,6 +687,12 @@
   }
 }
 
+.mdc-text-field--textarea:not(.mdc-text-field--upgraded) {
+  .mdc-text-field__input {
+    transition: mdc-text-field-transition(border-color);
```
[the remark this answers is not in the document]
Comment (bonniezhou) [reply packages/mdc-textfield/mdc-text-field.scss]: Oh I think you're right, removed all transitions for border.

**Pull request merged**